packages/components/organisms/f-cookie-banner/src/components/CookieBanner.vue

```diff
@@ -0,0 +1,410 @@
+                        data-test-id="cookieBanner-close-button"
+                        aria-label="Close"
+                        @click="hideLegacyBanner" />
+                </div>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script>
+import { globalisationServices } from '@justeat/f-services';
+
+import ButtonComponent from '@justeat/f-button';
+import '@justeat/f-button/dist/f-button.css';
+import tenantConfigs from '../tenants';
+
+export default {
+    name: 'CookieBanner',
+
+    components: {
+        ButtonComponent
+    },
+
+    props: {
+        locale: {
+            type: String,
+            default: 'en-GB'
+        },
+
+        isHidden: {
+            type: Boolean,
+            default: false
+        },
+
+        showLegacyBanner: {
+            type: Boolean,
+            default: false
+        },
+
+        cookieExpiry: {
+            type: Number,
+            default: 7776000
+        }
+    },
+
+    data () {
+        const locale = globalisationServices.getLocale(tenantConfigs, this.locale, this.$i18n);
+        const localeConfig = tenantConfigs[locale];
+        const theme = globalisationServices.getTheme(locale);
+        const hideBanner = false;
+        const legacyBanner = false;
+
+        return {
+            copy: { ...localeConfig },
+            theme,
+            hideBanner,
+            legacyBanner
+        };
+    },
+
+    watch: {
+        isHidden (newVal) {
+            if (newVal) {
+                this.hideBanner = true;
+            } else {
+                this.hideBanner = false;
+            }
+        },
+        showLegacyBanner (newVal) {
+            if (newVal) {
+                this.legacyBanner = true;
+            } else {
+                this.legacyBanner = false;
+            }
+        }
+    },
+
+    mounted () {
+        this.checkLegacyBannerFlag();
+        this.checkCookieBannerCookie();
+        this.focusOnTitle();
+    },
+
+    methods: {
+        /**
+         * Actions for "Accept all cookies" button
+         */
+        acceptActions () {
+            this.setCookieBannerCookie('full');
+            this.setLegacyCookieBannerCookie();
+            this.dataLayerPush('full');
+            this.hideBanner = true;
+        },
+        /**
+         * Actions for "Accept only required cookies" button
+         */
+        nonAcceptActions () {
+            this.setCookieBannerCookie('necessary');
+            this.setLegacyCookieBannerCookie();
+            this.dataLayerPush('necessary');
+            this.resendEvents();
+            this.removeUnnecessaryCookies();
+            this.hideBanner = true;
+        },
+        /**
+         * Set focus to the cookie consent banner title for accessibility
+         */
+        focusOnTitle () {
+            this.$refs.cookieBannerHeadingFocus.focus();
+        },
+        /**
+         * hide legacy banner
+         */
+        hideLegacyBanner () {
```

Comment:
If we have this method, then we should call it in more places. We have several `this.hideBanner = true` instead of calling this function.

Reply:
Removed the method and set `hideBanner` directly on the close button